vpr/src/base/read_blif.cpp

```diff
@@ -249,14 +268,19 @@ struct BlifAllocCallback : public blifparse::Callback {
             const t_model* blk_model = find_model(subckt_model);
 
             //We name the subckt based on the net it's first output pin drives
+            //We also check that the output nets don't have other drivers already.
             std::string subckt_name;
+            bool subckt_name_set = false;
             for(size_t i = 0; i < ports.size(); ++i) {
                 const t_model_ports* model_port = find_model_port(blk_model, ports[i]);
                 VTR_ASSERT(model_port);
 
                 if(model_port->dir == OUT_PORT) {
-                    subckt_name = nets[i];
-                    break;
+                    if (subckt_name_set == false) {
+                        subckt_name = nets[i];
+                        subckt_name_set = true;
+                    }
+                    assert_net_has_no_driver(nets[i]);
```

Comment:
Likely best to move this check down to after we have the actual `net_id` from `create_net()`. We can still figure out if it's an output by looking at `pin_type`.